### ISSUE TITLE
JS class privacy tag syntax error injected upon file save

### ISSUE THREAD

**Issue by dshipp25** [microsoft/vscode#129787]
Issue Type: <b>Bug</b>

When the file is saved, the class private Hash tag is separated from it's identifier, introducing a syntax error. This code demonstrates:

```js
class testthis {
  #privateProperty;
    constructor() {
        this.#privateProperty = 0;
    }
    #privateMethod() {
        return this.#privateProperty;
    }
};
```

Upon saving the file, it becomes

```js
class testthis {
    #
    privateProperty;

    constructor() {
        this.#privateProperty = 0;
    }

    #
    privateMethod() {
        return this.#privateProperty;
    }
}; which causes an error during code interpretation - identified missing after #.
```

VS Code version: Code 1.58.2 (c3f126316369cd610563c75b1b1725e0679adfb3, 2021-07-14T22:10:15.214Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 x 2803)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.65GB (8.02GB free)|
|Process Argv|--crash-reporter-id 9d98c34d-dc9e-4de9-96e9-e012074fac71|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none (all disabled to retest error...)

**Comment by IllusionMH** [comment]
/needMoreInfo

Are you using built-in TS version (used for JS as well) or workspace version?

VS Code 1.58.2 ships with VS Code that doesn't produce this result.

> Extensions: none (all disabled to retest error...)

You used command line or <kbd>F1</kbd> and `>Developer: Reload Window With Extensions Disabled`? Some extensions require window reload to be completely diusabled?

Do you have any actions configured in case e.g. `editor.codeActionsOnSave`? Do you have any watchers/formatters outside of VS Code?

**Comment by dshipp25** [comment]
Thank you for such a quick response. I'm new to Visual Studio Code so I can't answer many of your questions properly (TS version or workspace, I'm not sure...) I'm am using v1.58.2 so your comment makes me think I didn't properly disable the extensions. I have not installed any watcher/formatters myself other than an extension that uses Beautify.

Before my message, I just clicked disable on each extension and tried to verify they were disabled, but wasn't sure they really did. Just now, I used your F1 command and the problem did not reoccur. I then reenabled extensions, disabled or uninstalled all of them, reloaded VSCode and reinstalled one by one.  I found the culprit: JS-CSS-HTML Formatter.  I won't be using that one now.

Thank you for your guidance. Problem solved.
